Some map styling and enhancements

jszwedko opened the pull request:
Style events as small circles rather than the default LeafletJS markers.

Add locate.locatecontrol to the map to allow the user to locate
themselves. Useful if the user has panned and wants to return to their
location.

This changes the geolocation behavior on the start page to, instead of
doing the location and loading the map, simply load the map and ask the
map to locate the user (via the added locatecontrol). Though we wanted
to preserve the behavior of location before map load, I wasn't able to
get this to play nicely with rendering the user's location via the newly
added locate control which doesn't have an interface for setting the
current location (it depends on receiving a Leaflet.LocateEvent).
I Opened domoritz/leaflet-locatecontrol#209 to
get the maintainer's feedback on adding an interface to set the location.

jszwedko committed:
Style events as small circles

Rather than the default LeafletJS markers.

jszwedko committed:
Add map locate control

Add locate.locatecontrol to the map to allow the user to locate
themselves. Useful if the user has panned and wants to return to their
location.

This changes the geolocation behavior on the start page to, instead of
doing the location and loading the map, simply load the map and ask the
map to locate the user (via the added locatecontrol). Though we wanted
to preserve the behavior of location before map load, I wasn't able to
get this to play nicely with rendering the user's location via the newly
added locate control which doesn't have an interface for setting the
current location (it depends on receiving a Leaflet.LocateEvent).
I Opened https://github.com/domoritz/leaflet-locatecontrol/issues/209 to
get the maintainer's feedback on adding an interface to set the location.

akegan commented:
I think they're a little hard to pick out, looks like the update also didn't make it through to the event page 
![image](https://user-images.githubusercontent.com/11825994/44940281-3cd6f500-ad41-11e8-9388-46c79992fc85.png)

akegan commented:
something is also funky about the alignment on the bottom of the event view 
![image](https://user-images.githubusercontent.com/11825994/44940301-95a68d80-ad41-11e8-9168-2dbf02a505c4.png)

akegan commented:
I also noticed that the circles and the user's location look pretty similar. Other than that, the code looks good!

jszwedko commented:
@akegan thanks for reviewing! To your points:

* Yeah, Scott is still working on styling those buttons so they are a bit wonky at the moment
* 👍 I agree the user location and events look similar; I figured I'd let Scott play with the styling once the structure was in place
* I initially thought using the default icon on the event view page would be fine, but, agreed, it'd be better if they were consistent. Will update!

jszwedko committed:
Render events the same on their event page as on the map

Pull out rendering of points as circles into a new component,
EventFeature, that can be used on both the EventMap and Event
components.

jszwedko commented:
@akegan pulled out rendering of events so that they look the same on the map and on the event page: https://github.com/SFDigitalServices/neighborhood-noticing/pull/13/commits/a1e3ec363bce0891ff1f2b25b578e4d00d6159f0

jszwedko committed:
Remove stray console.log

akegan commented:
looks good with changes. I'll hold off on style critique until scott finishes his updates.

My only comment is that I'm not immediately sure what an EventFeature is. Maybe EventPin? It's getting hard to come up with more Event{noun} combos..

akegan reviewed (approved):
Approving, but I think a different name for EventFeature, or even some more documentation in the file would be helpful!

jszwedko commented:
@akegan 👍 yeah, I struggled with the name. I thought about `EventPin`, but the events can also be polygons and lines which led me back to the GeoJSON terminology of a "Feature" which encompasses all of these geometries.

I'll add some documentation though, good call.

jszwedko committed:
Add documentation to EventFeature

jszwedko commented:
Added documentation comment in https://github.com/SFDigitalServices/neighborhood-noticing/pull/13/commits/9048f1cb18413ea8b6a72a5ed8b6081fd671463b
